.github/workflows/strict-type-check.yml

```diff
@@ -0,0 +1,27 @@
+name: type check
+
+on:
+  pull_request:
+    paths:
+      - "**.py"
+  push:
+    paths:
+      - "**.py"
```

Comment:
this should also trigger for `.github/type_check.sh`, `.github/pyright-config.json`, `.github/workflows/strict-type-check.yml` and on `**.pyi`